hpx/runtime/agas/big_boot_barrier.hpp

```diff
@@ -82,19 +83,38 @@ struct HPX_EXPORT big_boot_barrier : boost::noncopyable
     parcelset::locality here() { return bootstrap_agas; }
     parcelset::endpoints_type const &get_endpoints() { return endpoints; }
 
+    template <typename Action, typename... Args>
```

Comment:
I think this leads to a better API. It hides the need for transfer_action and unique_ptr. You pass in the action by value and it's arguments. The transfer action will then be constructed here directly instead of at the calling site. Things like potential small buffer optimizations can now be easily added to parcel without breaking any dependant code.